### ISSUE TITLE
notes on strategy, architecture and training

### ISSUE THREAD

**Issue by stnava** [alex-reardon/siqc#3]
**general idea** 

train a regression/classification model to predict known classes of reduced image quality 

functions in this package will be applied to a "good" input image  and will produce an output image with known failure class(es)

as many of these images as possible  should be generated and written to disk 

each image should:

* be converted to [0,1] range
* be mapped to a consistent resolution ( maybe 2x2x2 isotropic -- this should be a variable that one can change ) ... this can be controlled by a template space but be careful about interpolation
* then be artificially corrupted where you keep track of the type(s) of corruption applied to each 
* another consideration is geometric augmentation -- some rotation/shearing should be applied but not too much unless the rotation is a corruption class ( ie upside down brains etc )
* write to disk a csv file which has  the input_image_name, output_image_name, corruption_1 ... corruption_n    ie which types of corruption have been applied 
* should also generate ample examples with side by side visual comparison of the input and output images along with the class(es) of corruption 
* recall:  should fairly and sensibly sample both modality and corruption class to prevent bias in the resnet rater

these should then be converted to numpy arrays of a machine readable size which can then be indexed rapidly on the fly during training ... 

so the idea is to *first* generate the training data ( not worrying about any deep learning stuff explicitly ) ... then convert to arrays ... then start training 

at first pass, you could start with maybe 1000 training images s.t. you can generate an end-to-end pipeline and get an initial model going ... then add to this later eg 10,000 or 100,000 or more examples for a more refined model.     should keep some "real" data on hand to test with ... 

**architecture** 

* recommend the resnet model used in the T1qc example
* it would be possible to initialize this model with the T1qc weights in order to accelerate training
* the last layer should match the number of classes you will predict
* should be either softmax or sigmoid (maybe start with softmax) but see training notes below

**training**

* losses:
   * start with MSE or MAE
   *  when starting to converge, switch to CCE or something similar $+$ Pearson ( if achievable ) as in the T1qc example
* augmentation
    * should use fixed data written to disk so it is possible to train rapidly by just reading nparrays 
* use 10% left out data (or whatever is reasonable given memory constraints ) to determine performance as training goes on

**other notes** 

write an inference function right away so that you can understand how this will be applied in the end application --- and to make sure you have the pre-processing well-documented and understood
